Better onboarding

zeitschlag opened the pull request:
see #690

zeitschlag committed:
Add pages and page-controller for better onboarding (#690)

zeitschlag committed:
Add localization to Xcode-project (#690)

This way you can easily open `app.json` in Xcode

zeitschlag committed:
Add styling and localization to education (#690)

zeitschlag committed:
Figure out a way to attach to scrolling in pages (#690)

zeitschlag committed:
Fix scrollview-offset-calculation (#690)

nemobis commented on Mastodon/Scene/Onboarding/Welcome/View/Pages/WelcomeContentPageView.swift:
```diff
@@ -0,0 +1,59 @@
+//  WelcomeContentPageView.swift
+//  Mastodon
+//
+//  Created by Nathan Mattes on 26.11.22.
```
Please use  the yyyy-mm-dd format.
https://xkcd.com/1179/

DK1DA replied:
We could also consider dropping the header in general. In its current form I don't see any value in it at all. The filename is part of, well, the filename. And the date of creation can be inferred from the Git history.

nemobis replied:
Definitely. I'm not sure what the status of copyright assignment is in this project, but *if* you want to have file-level headers this is the recommended way: https://reuse.software/faq/#step-2

zeitschlag committed:
[WIP] Move layoutcode and move plane (#690)

Turns out: Changing the constant of a layoutconstraint does the job, but is' laggy af, so I have to come up with another solution.

DK1DA commented on Mastodon/Scene/Onboarding/Welcome/View/Pages/WelcomeContentPage.swift:
```diff
@@ -0,0 +1,49 @@
+import UIKit
+import MastodonLocalization
+
+enum WelcomeContentPage: CaseIterable {
```
As this enum is, at least currently, only used to switch over different content with the same layout, I'm wondering if we should make this more generic e.g. `pageOne`, `pageTwo`, etc. then we'd only need to update the i18n values later on, if we'd like to change the onboarding content.

zeitschlag committed:
Make images move (#690)

They're not moving in the right direction yet, and it's entirely based on autolayout and not on layoutSubviews. We'll see, if that's the way to go.

zeitschlag committed:
[WIP] Fix animations on illustration-view (#690)

It's still broken on iPad.

zeitschlag committed:
Use native navigationItem instead of bottom-view (#690)

zeitschlag committed:
Apply new style to server-rules (#690)

zeitschlag committed:
Move files (#690)

zeitschlag committed:
Replace NavigationActionView with a view with a simple button (#690)

zeitschlag committed:
Get rid of title (#690)

zeitschlag committed:
Remove meta-information (#690)

zeitschlag committed:
Remove meta-information (#690)

zeitschlag committed:
Update checkmark to match the new design (#690)

zeitschlag committed:
(Temporarily) Remove search field from header (#690)

zeitschlag committed:
Add thumbnail image (#690)

thumbnail-download is still missing, it's just the UI-element

zeitschlag committed:
Get rid of highlight-indicator (#690)

zeitschlag committed:
Show back-button on server-list (#690)

kimar committed:
Adjust onboarding cloud alpha

zeitschlag committed:
Download thumbnails (#690)

zeitschlag committed:
Minor styling (#690)

zeitschlag committed:
Some styling in pick-server-header (#690)

zeitschlag committed:
Re-enable search (#690)

zeitschlag committed:
Style server-rules-tableview (#690)

kimar committed:
Remove sign-in button shadow

zeitschlag committed:
Add basic classes for privacy-screen during onboarding (#690)

kimar committed:
Implement mastodon wordmark on WelcomeContentPage

kimar committed:
Remove comment

zeitschlag committed:
Show privacy-policies, connect screens (#690)

zeitschlag committed:
Add some localization here and there (#690)

zeitschlag committed:
Add explanatory text to privacy screen (#690)

This also includes a refactor on the ServerRules-screen. It's built the same way

zeitschlag committed:
Show large titles in onboarding (#690)

zeitschlag committed:
Set new title for account-creation (#690)

zeitschlag committed:
Fix some colors (#690)

zeitschlag committed:
Re-enable loading-indicator for server-picker (#690)

zeitschlag committed:
Set color for privacy-cells (#690)

Also: Make light blurple the dark-mode-version of blurple

zeitschlag committed:
Remove avatar-selection (#690)

zeitschlag committed:
Adjust registration-screen to match updated design (#690)

Please note: this doesn't contain any logic, yet. It's solely UI.

zeitschlag committed:
Show spinner when registering (#690)

zeitschlag committed:
Fix localization (#690)

zeitschlag committed:
Fix typo/build (#690)

zeitschlag committed:
Consider password-confirmation on signup (#690)

This is now the logic-part. The passwords must be the same to register.

zeitschlag committed:
Download languages (#690)

zeitschlag committed:
Add chevron (#690)

spoiler: for selecting a language and registration-status

zeitschlag committed:
Add language-selector (#690)

For now, it searches for German servers only because.

zeitschlag committed:
[WIP] Add language-selection to onboarding (#690)

Consider this WIP, for now, languages are hard-coded

zeitschlag committed:
Make chevron smaller (#690)

zeitschlag committed:
Use real languages for language-selection (#690)

zeitschlag committed:
Remove obsolete comment (#690)

zeitschlag committed:
Add strings for signup-speed (#690)

zeitschlag committed:
Add button for signup-speed-filter (#690)

Filtering should work, too, but: no menu yet, so: it's a bit useless 😅

zeitschlag committed:
Move "all"-categories behind filters (#690)

zeitschlag committed:
Show menu for signup-speed-selection (#690)

zeitschlag committed:
Fix button-title (#690)

zeitschlag committed:
Fix search-bar-placeholder for server-picker (#690)

zeitschlag committed:
Make sign-up-speed-filter work (#690)

It took me a while to figure out that indexedServers were already filtered. To some degree it’s just fighting people’s inability to read…

zeitschlag committed:
Fix build :see_no_evil:

this is a bit embarrasing, I guess :D

zeitschlag committed:
Remove all from ServerPicker (#690)

kimar committed:
chore(onboarding): Improve airplane movement add clouds movement

kimar committed:
chore(onboarding): Update hill images and animation

zeitschlag committed:
Add username-suggestion (#690)

zeitschlag committed:
Localize username-suggestion (#690)

Please be aware of the username-requirements, when changing the localized versions.

zeitschlag committed:
Fix email-confirmation-screen-title (#690)

what a word.

kimar committed:
chore(onboarding): Decrease spacing between title and top of WelcomeViewController

zeitschlag committed:
Remove navigation-view on email-confirmation (#690)

zeitschlag committed:
Fix eh wrong eh translation?

zeitschlag committed:
Add button to resend email (#690)

Make it work (<- we're here)
Make it right
Make it fast

zeitschlag committed:
Localize and style resend-button (#690)

zeitschlag committed:
Add missing localization (#690)

zeitschlag committed:
Randomly select a server for user (#690)

zeitschlag committed:
Fix scrolling-issue (#690)

zeitschlag committed:
Some polish here and there in regards (#690)

Mostly navigationbar-related.

zeitschlag committed:
Fetch all servers for login-auto-complete (#690)

so far, only those with an open registration were available. Server needs to be adjusted as well.

zeitschlag committed:
Fix warning

zeitschlag committed:
Revert "Some polish here and there in regards (#690)"

This reverts commit a84910f2aba114a6600337dd9d50d3d5e5c749c6.

zeitschlag committed:
Show menu right away (#690)

This was a nice bug: The menu for signup-speed/language was attached too late. The entire button-delegate-stuff was obsolete in the end

zeitschlag committed:
Fix language-based random server-selection (#690)

During testing on device I noticed, that "next" didn't work. The reason was that I used the wrong language code. In case there's no server in my language, the server gets totally randomly selected.

zeitschlag committed:
UX-improvements for server-picker (#690)

Sprinkle in some haptic feedback and show menu immediately

zeitschlag committed:
Fix text-color for education-screens (#690)

In dark mode, the text was white and so unreadable

zeitschlag committed:
Add a blurry background (#690)

zeitschlag committed:
Make blurry-view lighter (#690)

zeitschlag committed:
Replace the pagecontrol with a collection view (#690)

Scrolling wasn't smooth with pageviews, as they do some black magic with scrollviews (like resetting contentOffset). If you depend on contentOffset, this breaks things and makes them hard.

zeitschlag committed:
Fix content-page-size (#690)

zeitschlag committed:
Add page control (#690)

zeitschlag committed:
Fix title for login-screen (#690)

Was a norma label, is the _real_ title now

zeitschlag committed:
Remove duplicated code (#690)

zeitschlag committed:
Fix localization-json-issues (#690)

zeitschlag committed:
Remove obsolete dependency (#690)

zeitschlag committed:
Minor cleanup here and there (#690)

DK1DA commented on Localization/StringsConvertor/input/Base.lproj/app.json:
```diff
@@ -287,9 +300,19 @@
+            "title": "Privacy",
+            "policy": {
+                "ios": "Privacy Policy - Mastodon for iOS";
+                "server" = "Privacy Policy - %s";
```
There is still some invalid JSON here.

zeitschlag replied:
Ah, shit, thank you for fixing it!

kimar reviewed (approved):
LGTM besides this one broken JSON 👏  Great work!

kimar committed:
chore(i18n): fix broken json

kimar committed:
chore(i18n): Update strings

kimar committed:
Revert "chore(i18n): Update strings"

This reverts commit be3b91e334c940d41d39aefa266dbe4a15bc1e89.
